src/server.ts

```diff
@@ -1,8 +1,73 @@
 import express, { Request, Response } from 'express';
 
+import { Prisma, PrismaClient, User } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
 export const server = express();
+
+type ErrorMessage = {
+  message: string;
+};
+
+type EmptyObj = Record<string, never>;
+
+// Middlewares
+server.use(express.json());
+
+// Endpoints
 server.get('/', (_: Request, res: Response) => {
   res.send({
     message: 'hello world',
   });
 });
+
+// User endpoints
+server.get(
+  '/users',
+  async (
+    _: Request<EmptyObj, User[], EmptyObj>,
+    res: Response<User[]>
+  ): Promise<Response<User[]>> => {
+    const users = await prisma.user.findMany();
+    return res.send(users);
+  }
+);
+
+interface NewUserRequestBodyParams {
+  uid: string;
+  username: string;
+}
+
+server.post(
+  '/users/new_user',
+  async (
+    req: Request<EmptyObj, User, NewUserRequestBodyParams>,
+    res: Response<User | ErrorMessage>
+  ): Promise<Response<User | ErrorMessage>> => {
```

Comment:
I was hoping to get strong typing via protobufs or something of that nature to this API so that clients of the API had predictable interfaces.  But since that would require learning that on TOP of learning an ORM / brushing up on PostgreSQL, I went with just having strong typings at the request / response level